### ISSUE TITLE
sumtype invalid initializer

### ISSUE THREAD

**Issue by LouisSchmieder** [vlang/v#5688]
**V version:** V 0.1.28 bdecbc1.1416c70

**OS:** Manjaro

**What did you do?**
```v
type Abc = int | string

fn main() {
    a := Abc(1)

    match a {
        int {
            println(a)
        }
        else {}
    }
}
```

**What did you expect to see?**
`1`
 
**What did you see instead?**
```/home/daniel/.cache/v/main.tmp.c: In function ‘main__main’:
/home/daniel/.cache/v/main.tmp.c:9451:16: error: invalid initializer
 9451 |  main__Abc a = 1;
```

**Comment by danieldaeschle** [comment]
This is a c error, it is a bug.

**Comment by joe-conigliaro** [comment]
fixed in #d82e6c9cd95c8e5a307cd84a9ba54e06eca77b6f
